pallene/driver.lua

```diff
@@ -133,33 +134,42 @@ function driver.compile(argv0, input_ext, output_ext, input_file_name)
 
     local mod_name = string.gsub(base_name, "/", "_")
 
-    local first_step = step_index[input_ext]  or error("invalid extension")
-    local last_step  = step_index[output_ext] or error("invalid extension")
-    assert(first_step < last_step, "impossible order")
-
-    local file_names = {}
-    for i = first_step, last_step do
-        local step = compiler_steps[i]
-        if (i == first_step or i == last_step) then
-            file_names[i] = base_name .. "." .. step.name
-        else
-            file_names[i] = os.tmpname()
+    local ok, errs
+    if output_ext == "lua" then
+        assert(input_ext == "pln")
+        local ast, errs, input = driver.compile_internal(input_file_name, "checker")
```

Comment:
This `errs` is shadowing the other errs so when you do a return `ok, errs` at the end of the function we won't have the actual error message.

This reminds me that we probably should add a test case to ensure that type errors are properly reported to the user if we call `--emit-lua` in a file that has type errors.